gitium/inc/class-gitium-submenu-status.php

```diff
@@ -94,29 +94,59 @@ public function save_ignorelist() {
 	}
 
 	public function save_changes() {
-	    $gitium_save_changes = filter_input(INPUT_POST, 'GitiumSubmitSaveChanges', FILTER_SANITIZE_FULL_SPECIAL_CHARS);
+		$gitium_save_changes = filter_input(INPUT_POST, 'GitiumSubmitSaveChanges', FILTER_SANITIZE_FULL_SPECIAL_CHARS);
 		$gitium_commit_msg = filter_input(INPUT_POST, 'commitmsg', FILTER_SANITIZE_FULL_SPECIAL_CHARS);
-	    if ( ! isset( $gitium_save_changes ) ) {
+		if ( ! isset( $gitium_save_changes ) ) {
 			return;
 		}
 		check_admin_referer( 'gitium-admin' );
-
+	
 		gitium_enable_maintenance_mode() or wp_die( __( 'Could not enable the maintenance mode!', 'gitium' ) );
+	
+		// Check if the local repository is behind the remote
+		$behind_commits = $this->git->get_behind_commits();
```

Comment:
`get_behind_commits` runs `git rev-list` which comparse only commits that exists localy. In order to check comits on remote you have to run a `git fetch` first, there is a function that I think does that `fetch_ref()`.

Reply:
You can inspire from `gitium_merge_and_push` function and call `merge_with_accept_mine` but there is a lock that it should be aquired. You can try caling `gitium_merge_and_push` directly.

Reply:
Now I see that `gitium_merge_and_push` is called below at line 143.

What is then issue that you try to fix? Do you have some STR, or maybe some errors messages ? 
Maybe only calling gitium_disable_maintenace_mode is enough?

I see that this function `save_changes` only when `Save changes` button is pressed? Where problems with this?